pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/controller/PinotControllerTransportFactory.java

```diff
@@ -80,6 +81,7 @@ public PinotControllerTransportFactory withConnectionProperties(Properties prope
             DEFAULT_CONTROLLER_CONNECT_TIMEOUT_MS));
     _handshakeTimeoutMs = Integer.parseInt(properties.getProperty("controllerHandshakeTimeoutMs",
             DEFAULT_CONTROLLER_HANDSHAKE_TIMEOUT_MS));
+    _appId = properties.getProperty("appId", "");
```

Comment:
Same here, we want to set `null` when it is not configured
```suggestion
    _appId = properties.getProperty("appId");
```